utils/fasm/CMakeLists.txt

```diff
@@ -0,0 +1,51 @@
+cmake_minimum_required(VERSION 2.8.12)
+
+project("genfasm")
+
+#Create library
+add_library(fasm
+  src/fasm.cpp
```

Comment:
Using GLOB_RECURSE for including sources is actually an anti-pattern, especially with CMake, and should be avoided.

From CMake's own documentation:
```
(We do not recommend using GLOB to collect a list of source files from your source tree. If no CMakeLists.txt file changes when a source is added or removed then the generated build system cannot know when to ask CMake to regenerate.)
```

Reply:
I guess I respectfully disagree with CMake's reasoning. Avoiding GLOB_RECURSE just forces the user to manually edit the CMakeLists.txt to inform the build system of the dependency change -- which is more work (and probably more error prone) than just re-calling CMake.

To me, keeping the human out of that loop seems more beneficial.
It also ensure the CMakeLists.txt doesn't get out of sync with the actual directory/file structure, and that all the code actually gets compiled (no orphan files that were accidentally left out of the list, or not updated in the list after a source file rename).

Since we do use GLOB_RECURSE else where my preference (for consistency) would be to use it here too. But its not a big issue either way, so if you strongly prefer it this way we can leave it as is.

Reply:
> I guess I respectfully disagree with CMake's reasoning. Avoiding GLOB_RECURSE just forces the user to manually edit the CMakeLists.txt to inform the build system of the dependency change -- which is more work (and probably more error prone) than just re-calling CMake.

You are solely thinking about the developer make the change, not the other developers that consume the change.  If 1 developer adds a file, and 50 have to re-run CMake after getting a link error because of the GLOB not finding an added .cpp, how is that a fair balance?

Anyways my preference is to be explicit.